### PR TITLE
storage: Improve side panel layout, again

### PR DESCRIPTION
**Pull request by mvollmer**
This uses Patternfly `Flex` components to do the layout, and CSS for font sizes and margins.

image

### CODE REVIEW

File: pkg/storaged/storage.scss
Comment (garrett) [inline pkg/storaged/storage.scss:642]: This variable will change soon in PR https://github.com/cockpit-project/cockpit/pull/15183 (which removes a _lot_ of Cockpit-specific CSS variables and standardizes the remaining ones with a prefix)

Specifically, to `--ct-color-link-hover-bg`
https://github.com/cockpit-project/cockpit/pull/15183/files#diff-7d0ccb2f6de8d81d3e61bc5c350c8b0461851c05480bee0f5d715d4ee8001909R309

I guess I need to finish up that PR. :wink:

File: pkg/storaged/storage.scss
Comment (garrett) [inline pkg/storaged/storage.scss:633]: This is better with shorthand:

```suggestion
    padding: var(--pf-global--spacer--sm) var(--pf-global--spacer--md);
```

**Review by garrett — CHANGES_REQUESTED** [review]
We'll have to change the color at some point and should change the padding to shorthand. I still need to review this locally; right now, I just glanced at the diff (mainly focusing on the CSS).

File: pkg/storaged/storage.scss
Comment (garrett) [reply pkg/storaged/storage.scss:642]: Note: This shouldn't be updated until that lands. Else, it should stay the same and be updated in that other PR. (Whichever PR lands second of these two will need to change it, in other words.)

Comment (mvollmer) [reply pkg/storaged/storage.scss:642]: Can I use that prefix already here?

File: pkg/storaged/storage.scss
Comment (mvollmer) [reply pkg/storaged/storage.scss:633]: Done!

Comment (garrett) [reply pkg/storaged/storage.scss]: Explanation:

CSS has shorthand to combine things, it takes the form of one of the following:

- analogue clock directions: top right bottom left (this is why it has funky ordering... just think of the hands moving around a clock and you'll be fine)
- top/bottom left/right (what I'm using above)
- top sides bottom
- every-direction-at-once

It's only useful when defining all the padding-* border-* margin-* etc. at one time. (You can't skip a side, so re-declaring something will overwrite it.)

File: pkg/storaged/storage.scss
Comment (KKoukiou) [reply pkg/storaged/storage.scss:639]: > We'll have to change the color at some point and should change the padding to shorthand. I still need to review this locally; right now, I just glanced at the diff (mainly focusing on the CSS).

@garrett What does it mean we 'have to change the color'? You mean what table-hover used to do? If that's the case, isn't having links enough?

**Review by KKoukiou — APPROVED** [review]
I test it to me it looks good, and the code it much more PF4isch than before. 
I don't really understand what is the blocker here, I will wait for monday @garrett's feedback.

**Comment by garrett** [comment]
I've been trying to figure out how to make warning icons and nothing I do seems to trip it up.

How can I reliably make warning icons appear for this page (in a VM)?

**Comment by mvollmer** [comment]
> How can I reliably make warning icons appear for this page (in a VM)?

An easy way is to mount a filesystem in Cockpit, and then unmount it in the terminal.

**Review by garrett — APPROVED** [review]
LGTM

Notes:

- I've tried to figure out a way to recreate the issue pointed out in the other PR @ https://github.com/cockpit-project/cockpit/pull/15261#issuecomment-782085350 but cannot.

- It's a little weird to have an unnamed div in the panel for the decoration, but I guess it's there because of the flex. It would be great if it had a name like .sidepanel-row-deocration, but as it doesn't have any custom CSS (at least at the moment?) it's probably not really a big deal. I understand that it's there because its parent element is a flex container. So it's not really an issue.

File: pkg/storaged/storage.scss
Comment (garrett) [reply pkg/storaged/storage.scss:639]: @KKoukiou it means to change the name of the variable; it's changing in PR #15183

**Pull request merged**
